### DISCUSSION TITLE
updates, issues and discussions should be idenfied without opening each post

### DISCUSSION THREAD

**Discussion by ineuw** [autokey/autokey#692] [Maintenance goals]
Is there a way to include the Autokey version of an issue, or a git merge, in the title of the notice?

**Comment by josephj11** [comment]
IDK, but they could be manually labeled somehow (by me and the devs ...).

You'd still have to open them to see the labels though. Maybe something could be added to the title, but new issues don't usually have labels yet because most users don't have permission to assign them.

@Elliria might know if there is anything in YAML that could be added to our Issue template, but that still wouldn't affect discussion posts.

What benefit are you looking for? We do get such posts on a regular basis, but the traffic volume is relatively low.

As far as labels go:

We currently use the Beta, Development, and 0.90.4 labels. Thankfully, we haven't needed that last one for a while.

Issues need to be divided into production (assumed by default), development, and beta. I'm not sure what further distinctions would be useful (other than open/closed). There's also Won't Fix and Upstream, but they seem a bit tangential to your concerns.

I haven't been labeling discussions or PRs, but I could certainly do that if desired.

Related:
I use a chromium add-on, Helper for GitHub Notifications in Gmail, which adds a View Issue on GitHub button to each such post and I usually click on that immediately and skip reading the  email version, but I still have to open the emails to get to that button. It used to be available when hovering on an email header in the list of emails, but Google updated something that broke that feature.

**Reply by ineuw** [reply]
Thanks, I appreciate that you recognize that it wold be a step forward, but I don't expect a solution, other than adding the version to the title of the post. 

@Elliria, I am sure that you have more important issues to deal with. This is not that important, another solution is to search the message body for a key word.

**Reply by ineuw** [reply]
One quick fix to identify a post, is to remind this page's header.
Another would be that a post could not be saves without a version tag. Instead "autokey-gtk", it could be gtk-95, gtk-96b, qt-95.1 . . etc.

**Reply by josephj11** [reply]
I think that combining two separate attributes into one tag is a bad idea. I may need to know about issues of a version independent of the front end in use and vice versa.

If you were sufficiently masochistic :) , you could write a really large AutoKey script to read each message for you and tell you which ones you want to look at. That might be ridiculously complex to get right and very fragile with respect to website changes.

Barring that, there are a few alternatives:

There is a search field at the top of the Discussions page and another one that takes a bunch of predicates at the top of the issues page. These work pretty well once you figure them out. 

There is also the browser search _site:_ predicate which is quite useful, but hard to use effectively on multipurpose/multi-project websites like GitHub or KDE.

Neither of these intersect with the notification emails we receive, but they are way better than nothing.

You can also choose to unwatch any issues (and probably discussions) that you no longer want to be informed about.

**Reply by ineuw** [reply]
I would be happy with any tag configuration. Joint, and/or separate tags. Just creating this thread should bring it to the attention of those who are interested.

Anything that would provide important details of a post at a glance, would improve our ability to provide a solution to problems which we know about.

**Comment by josephj11** [comment]
Keep in mind that very few people read things on our GitHub pages and probably even less look at the Discussions. IDK, but I suspect there's a significantly larger audience on Gitter  (and maybe even more on our Google Group, but this is probably a bit too technical for that group.)

Also, since tags are already a feature of issues, I'm happy to add and maintain a few more tags if there is general agreement as to what would be useful.

What I don't want to do is end up tagging something as 0.95.10. If it's still not fixed in 0.96, what do I do, retag it? Automatic/mass retagging tends to delete old tags, so the issues would have to be done one by one. Things like that could generate a lot of overhead work with little or no additional benefit. So we have to be careful to do things that are flexible enough to keep working as development continues.

**Comment by Elliria** [comment]
Hey there. Sorry about that, guys. I was away on a vacation of sorts in which I didn't take computering with me. My understanding, from everything I was able to find on how to use YAML with GitHub issue templates, is that we have two choices. We can leave the title blank and leave it up to the user to type one up, which is what we currently do, or we can insert one for them. The one we can insert would be the same one for each bug report and not one that uses a placeholder to insert dynamic user-input. That's actually a great idea, though, and would be a wonderful suggestion for either GitHub or YAML -- whichever of the teams would be most likely to implement it.

**Comment by Elliria** [comment]
We do, however, have full control over the form itself and we could make the version question the first one the user answers so that it's at the top and can be quickly checked. You'd still have to open the form to get to it, though.

**Comment by ineuw** [comment]
Adding the version number is also insufficient. I am dropping the idea.

A simpler and better idea would be a text search of the posts, **with the ability to search within the found posts.**

**Reply by josephj11** [reply]
If you have (copies of) your email stored locally, recoll will index all of that along with the rest of your system depending on how you configure it. It allows quite sophisticated searching and previewing of results. I don't use it very often, but it's a lifesaver when I need it. It routinely finds things Thunderbird's stupid search says aren't there. Of course, you have to make sure the message bodies have been downloaded (IMAP - sometimes, only the message headers are downloaded until you actually open the messages ) or they won't be there to search.

And, as stated above, GitHub also provides a pretty sophisticated issue search utility.

**Comment by ineuw** [comment]
Thanks. I don't keep many posts and prefer to search directly in github. It's time for me to test it.

**Comment by josephj11** [comment]
See [this](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for how it works.

**Comment by Elliria** [comment]
That GitHub search page is great, @josephj11. It's worthy of a bookmark for future reference considering how many ways there are to use it. Great find.
